compiler/src/codegen/compcore.re

```diff
@@ -278,56 +92,53 @@ let init_codegen_env = name => {
   },
   backpatches: ref([]),
   required_imports: [],
+  global_import_resolutions,
+  func_import_resolutions,
+  compilation_mode: Normal,
 };
 
-let reset = () => {
-  reset_labels();
-  List.iter(
-    imp => imp.mimp_used = imp.mimp_mod == grain_env_mod,
-    runtime_imports,
-  );
+let linked_name = (~env, id) => {
+  Printf.sprintf("%s_%d", id, env.dep_id);
+};
+
+let rec resolve_global = (~env, name) => {
+  switch (Hashtbl.find_opt(env.global_import_resolutions, name)) {
+  | Some(resolution) => resolve_global(~env, resolution)
+  | _ => name
+  };
 };
 
-let get_wasm_imported_name = (~runtime_import=true, mod_, name) => {
-  if (runtime_import) {
-    // Mark runtime import as used
-    Ident_tbl.find(runtime_imports_tbl, name).mimp_used =
-      true;
+let rec resolve_func = (~env, name) => {
+  switch (Hashtbl.find_opt(env.func_import_resolutions, name)) {
+  | Some(resolution) => resolve_func(~env, resolution)
+  | _ => name
   };
-  Ident.unique_name(name);
+};
+
+let reset = () => {
```

Comment:
What is the point of this function

Reply:
Many of our modules have a `reset` function that clears any internal state that calls whatever smaller reset functions there are. Did the same thing here.